src/index.ts

```diff
@@ -13,10 +13,18 @@ function main() {
     const canvas = <HTMLCanvasElement>document.getElementById("render-area");
 
     simulator = new Sim3D(canvas, {
-        world: {
+        defaultWorld: {
             xLength: 10,
             zLength: 20,
-            includeWalls: true
+            includeWalls: true,
```

Comment:
if we're going for a declarative approach, would it make more sense to take an array of walls? Maybe: 
```
`walls: Array<{xStart: number; ystart: number; xEnd: number; yEnd: number}>`
```

Reply:
Yeah I'm open to that. 
```typescript
interface WallSpec {
    xStart: number;
    zStart: number;
    xEnd: number;
    zEnd: number;
    thickness?: number;
    height?: number;
}

interface WorldConfig {
    xLength: number;
    zLength: number;
    walls?: WallSpec[]
}
```

So if `walls` is not provided, we don't draw any walls. Otherwise we use whatever walls we are given. As a convenience, if an empty array is provided, we'll just fill in the 4 perimeter walls.

How does that sound?

Reply:
I'm happy with that. I'll let others comment too

Reply:
I did a very rough implementation of a declarative SimWall... it's pretty nice!

```typescript
const newWallConfig: ISimWallDeclarativeSpec = {
            xStart: -1,
            zStart: -1,
            xEnd: 1,
            zEnd: 1,
            thickness: 0.05,
            height: 1,
            color: 0x0000ff
        };

        const newWall = new SimWallDeclarative(newWallConfig);
        this.scene.add(newWall.mesh);
```

Reply:
And this was the implementation:

```typescript
export class SimWallDeclarative extends SimObject {
    constructor(spec: ISimWallDeclarativeSpec) {
        super();

        const xDiff = spec.xEnd - spec.xStart;
        const zDiff = spec.zEnd - spec.zStart;
        const length = Math.sqrt((xDiff * xDiff) + (zDiff * zDiff));
        const midX = spec.xStart + (xDiff / 2);
        const midZ = spec.zStart + (zDiff / 2);

        let angleRad: number = 0;
        if (xDiff === 0) {
            angleRad = Math.PI / 2;
        }
        else if (zDiff === 0) {
            angleRad = 0;
        }
        else {
            angleRad = Math.atan(zDiff / xDiff);
        }

        const thickness = spec.thickness !== undefined ? spec.thickness : DEFAULT_THICKNESS;
        const height = spec.height !== undefined ? spec.height : DEFAULT_HEIGHT;
        const color = spec.color !== undefined ? spec.color : DEFAULT_COLOR;

        // We build the wall in the X direction, thickness in Z and height in Y
        // Then we rotate it appropriately, and shift it to (midX, 0, midZ)

        const meshGeom = new THREE.BoxGeometry(length, height, thickness);
        const wallMaterial = new THREE.MeshStandardMaterial({ color });

        const wallMesh = new THREE.Mesh(meshGeom, wallMaterial);

        // Set position
        wallMesh.position.x = midX;
        wallMesh.position.z = midZ;
        wallMesh.position.y = (height / 2);

        wallMesh.rotation.x = 0;
        wallMesh.rotation.z = 0;
        wallMesh.rotation.y = angleRad;

        this._mesh = wallMesh;
    }
}
```